hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogImpl.java

```diff
@@ -360,6 +366,12 @@ public DatanodeDeletedBlockTransactions getTransactions(
             transactions.getBlocksDeleted() < blockDeletionLimit) {
           Table.KeyValue<Long, DeletedBlocksTransaction> keyValue = iter.next();
           DeletedBlocksTransaction txn = keyValue.getValue();
+          if (firstProcessedTransactionKey == keyValue.getKey()) {
+            break;
+          }
+          if (firstProcessedTransactionKey == -1) {
+            firstProcessedTransactionKey = keyValue.getKey();
+          }
```

Comment:
We can reuse the `lastProcessedTransactionId` instead of introducing a local `firstProcessedTransactionKey` variable.

Reply:
Done